cgroup_poller.go

```diff
@@ -0,0 +1,99 @@
+package shh
+
+import (
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/heroku/shh/Godeps/_workspace/src/github.com/freeformz/filechan"
+)
+
+const (
+	CGROUPS_PATH = "/sys/fs/cgroup"
+)
+
+type Cgroup struct {
+	measurements chan<- Measurement
+	last         map[string]uint64
+	cgroups      []string
+	// The kernel will report CPU usage in centiseconds.  This
+	// stores the total centiseconds in the polling interval.
+	totalCentis uint64
+}
+
+func NewCgroupPoller(measurements chan<- Measurement, config Config) Cgroup {
+	return Cgroup{
+		measurements: measurements,
+		last:         make(map[string]uint64),
+		cgroups:      config.Cgroups,
+		// convert the interval to centiseconds
+		totalCentis: uint64(config.Interval.Nanoseconds() / 10000000),
+	}
+}
+
+func sanitizeMetricName(name string) string {
+	return strings.Replace(strings.Replace(name, ".", "-", -1), "/", "--", -1)
```

Comment:
I've never used it, but you might find that creating a [Replacer](https://golang.org/pkg/strings/#Replacer), statically, and then just calling it's `Replace` method reads better here.

There's also (FixupName)[https://github.com/heroku/shh/blob/master/utils.go#L197] in utils.go which might just do what you need here